### PR TITLE
Fix for tf.keras.regularizers.{l1,l2}(0.) with tf.get_variable

### LINKED ISSUE #22470: [keras compatibility] tf.keras.regularizers.l2() cannot be used in tf.get_variable()
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: custom code
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.10.1
- **Python version**: 3.6.4

### Describe the problem
`tf.keras.regularizers.l2()` cannot be used in `tf.get_variable()`. Currently I have to use `tf.contrib.layers.l2_regularizer()` in `tf.get_variable()`.  
Note that , `tf.keras.regularizers.l2()` seems to work well in `tf.layers.Dense()`.

In general, I expect that all `keras` functions like `regularizers`, `activations`, `initializers` could be used for `tensorflow`.

### Source code / logs
```
x = tf.get_variable(shape = [dim_in, dim_out],
    initializer=tf.keras.initializers.glorot_uniform(),
    regularizer=tf.keras.regularizers.l2(lambda))
```
Traceback:
>   File "/Users/mac/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 1467, in get_variable
>     aggregation=aggregation)
>   File "/Users/mac/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 1217, in get_variable
>     aggregation=aggregation)
>   File "/Users/mac/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 527, in get_variable
>     aggregation=aggregation)
>   File "/Users/mac/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 481, in _true_getter
>     aggregation=aggregation)
>   File "/Users/mac/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 930, in _get_single_variable
>     loss_name = loss.name
> AttributeError: 'float' object has no attribute 'name'

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue in #22470 where tf.keras.regularizers.{l1,l2}(l=0.) with tf.get_variable returns
```
AttributeError: 'float' object has no attribute 'name'
```

The issue only happened when regularization was disabled `l=0.`, as in that case, `regularization = 0.`
was returned directly.

This fix convert `regularization = 0.` to a tensor.

This fix fixes #22470.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by wt-huang** [comment]
@yongtang Could you look into the failing tests? Probably consider populating more test cases.

File: tensorflow/python/keras/regularizers.py [hunk 1]
```diff
@@ -54,7 +55,7 @@ def __init__(self, l1=0., l2=0.):  # pylint: disable=redefined-outer-name
     self.l2 = K.cast_to_floatx(l2)
 
   def __call__(self, x):
-    regularization = 0.
+    regularization = ops.convert_to_tensor(0., dtype=K.floatx())
```
Comment (fchollet) [inline tensorflow/python/keras/regularizers.py]: If l1=0 and l2=0 then we would be creating a useless node in the graph. So please place this inside a `if l1 or l2:`. Then please add some unit tests.

**Comment by yongtang** [comment]
Thanks @fchollet @wt-huang for the review. The PR has been updated, and a unit test has been added as well. Please take a look and let me know if there are any issues.

**Comment by wt-huang** [comment]
@yongtang Thanks for the fix! Made a couple of pylint changes.

**Comment by rspier** [comment]
@googlebot Ping

**Comment by wt-huang** [comment]
@yongtang For formality check, please confirm that you are okay with all the changes made so far. Thanks!

**Comment by yongtang** [comment]
@wt-huang  Yes I am ok with all the changes. Thanks for the help!

**Comment by yongtang** [comment]
@wt-huang Looks like all tests passed, except the copybara. Wondering if there is any update or anything else needs to be done?

**Comment by wt-huang** [comment]
Thanks @yongtang LGTM
@fchollet Could you take another look?

**Pull request merged**